Zend/zend_compile.c

```diff
@@ -4712,6 +4712,125 @@ static void zend_compile_ns_call(znode *result, znode *name_node, zend_ast *args
 }
 /* }}} */
 
+static zend_op *zend_compile_rope_add(znode *result, uint32_t num, znode *elem_node);
+static zend_op *zend_compile_rope_add_ex(zend_op *opline, znode *result, uint32_t num, znode *elem_node);
+static void zend_compile_rope_finalize(znode *result, uint32_t j, zend_op *init_opline, zend_op *opline);
+
+static zend_result zend_compile_func_sprintf(znode *result, zend_ast_list *args) /* {{{ */
+{
+	znode format_string;
+
+	if (args->children < 1) {
+		return FAILURE;
+	}
+
+	zend_compile_expr(&format_string, args->child[0]);
+	if (!(format_string.op_type == IS_CONST && Z_TYPE(format_string.u.constant) == IS_STRING)) {
```

Comment:
```suggestion
	if (format_string.op_type != IS_CONST || Z_TYPE(format_string.u.constant) != IS_STRING) {
```

Causes much less cognitive overhead for me. We usually use `||` for failure condition checks.